cos_registration_agent/cos_registration_agent.py

```diff
@@ -3,39 +3,110 @@
 import json
 import logging
 from pathlib import Path
-from typing import Set, Union
-from urllib.parse import urljoin
+from posixpath import join as urljoin
+from typing import Any, Optional, Set, Union
 
 import requests
 import yaml
 
 logger = logging.getLogger(__name__)
 
-API_VERSION = "api/v1/"
+API_VERSION = str("api/v1/")
 HEADERS = {"Content-Type": "application/json"}
```

Comment:
this HEADERS constant is used in some places, but now there is the self.headers member, should we rename it to HEADERS_CONTENT or something similar for clarity?

Reply:
Maybe I could rename it into `HEADERS_APPLICATION_JSON` ?

Reply:
Done